File: apps/python/blueprints/dependencies.py
import azure.functions as func
import json
import logging
import time

import requests as http_requests

bp = func.Blueprint()
logger = logging.getLogger(__name__)

EXTERNAL_URL = "https://httpbin.org/get"


@bp.route(route="dependencies/external", methods=["GET"])
def external_dependency(req: func.HttpRequest) -> func.HttpResponse:
    """Call an external HTTP service and return timing information."""
    logger.info("External dependency endpoint requested (url=%s)", EXTERNAL_URL)

    start_time = time.monotonic()
    try:
        response = http_requests.get(EXTERNAL_URL, timeout=10)
        elapsed_ms = round((time.monotonic() - start_time) * 1000)

        logger.info(
            "External call completed (status=%d, time=%dms)",
            response.status_code,
            elapsed_ms,
        )

        body = json.dumps({
            "status": "success",
            "statusCode": response.status_code,
            "responseTime": f"{elapsed_ms}ms",
            "url": EXTERNAL_URL,
        })
        return func.HttpResponse(body, mimetype="application/json", status_code=200)

    except http_requests.RequestException as exc:
        elapsed_ms = round((time.monotonic() - start_time) * 1000)
        logger.error(
            "External call failed (error=%s, time=%dms)", exc, elapsed_ms
        )

        body = json.dumps({
            "status": "error",
            "statusCode": 0,
            "responseTime": f"{elapsed_ms}ms",
            "url": EXTERNAL_URL,
            "error": str(exc),
        })
        return func.HttpResponse(body, mimetype="application/json", status_code=502)

File: apps/nodejs/src/functions/dependencies.js
const { app } = require('@azure/functions');

/**
 * Call an external HTTP service and return timing information.
 */
const EXTERNAL_URL = 'https://httpbin.org/get';

app.http('externalDependency', {
    methods: ['GET'],
    route: 'dependencies/external',
    handler: async (request, context) => {
        context.log(`External dependency endpoint requested (url=${EXTERNAL_URL})`);

        const startTime = Date.now();
        try {
            const response = await fetch(EXTERNAL_URL, { signal: AbortSignal.timeout(10000) });
            const elapsedMs = Date.now() - startTime;

            context.log(`External call completed (status=${response.status}, time=${elapsedMs}ms)`);

            return {
                status: 200,
                jsonBody: {
                    status: 'success',
                    statusCode: response.status,
                    responseTime: `${elapsedMs}ms`,
                    url: EXTERNAL_URL,
                },
            };
        } catch (err) {
            const elapsedMs = Date.now() - startTime;
            context.error(`External call failed (error=${err.message}, time=${elapsedMs}ms)`);

            return {
                status: 502,
                jsonBody: {
                    status: 'error',
                    statusCode: 0,
                    responseTime: `${elapsedMs}ms`,
                    url: EXTERNAL_URL,
                    error: err.message,
                },
            };
        }
    },
});
